letsencrypt-apache/letsencrypt_apache/configurator.py

```diff
@@ -322,17 +364,18 @@ def _add_servernames(self, host):
         :type host: :class:`~letsencrypt_apache.obj.VirtualHost`
 
         """
-        name_match = self.aug.match(("%s//*[self::directive=~regexp('%s')] | "
-                                     "%s//*[self::directive=~regexp('%s')]" %
-                                     (host.path,
-                                      parser.case_i("ServerName"),
-                                      host.path,
-                                      parser.case_i("ServerAlias"))))
-
-        for name in name_match:
-            args = self.aug.match(name + "/*")
-            for arg in args:
-                host.add_name(self.aug.get(arg))
+        # Take the final ServerName as each overrides the previous
+        servername_match = self.parser.find_dir(
+            "ServerName", None, host.path, exclude=False)
```

Comment:
it's hard to guess what `None` is for... it'd be more readable to say `arg=None` (though `arg` itself is not the best name itself)